Tasks/SqlAzureDacpacDeploymentV1/Utility.ps1

```diff
@@ -234,94 +246,93 @@ function Get-SqlPackageCommandArguments
-
-  if ($authenticationType -eq "aadAuthenticationPassword") {
-     $connectionString += @("Authentication=Active Directory Password; UID=$sqlUserName; PWD=$sqlPassword")
-  }
-  else {
-     $connectionString += @("Authentication=Active Directory Integrated;")
-  }
-
-  return $connectionString
+function Get-AADAuthenticationConnectionString {
+    param(
+        [String][Parameter(Mandatory = $true)] $authenticationType,
+        [String][Parameter(Mandatory = $true)] $serverName,
+        [String][Parameter(Mandatory = $true)] $databaseName,
+        [String] $sqlUserName,
+        [String] $sqlPassword
+    )
+
+    $connectionString = "Data Source=$serverName; Initial Catalog=$databaseName; "
+
+    if ($authenticationType -eq "aadAuthenticationPassword") {
+        $connectionString += @("Authentication=Active Directory Password; UID=$sqlUserName; PWD=$sqlPassword")
+    }
+    else {
+        $connectionString += @("Authentication=Active Directory Integrated;")
+    }
+
+    return $connectionString
 }
 
-function Execute-Command
-{
+function Execute-Command {
     param(
-        [String][Parameter(Mandatory=$true)] $FileName,
-        [String][Parameter(Mandatory=$true)] $Arguments
+        [String][Parameter(Mandatory = $true)] $FileName,
+        [String][Parameter(Mandatory = $true)] $Arguments
     )
 
     $ErrorActionPreference = 'Continue'
     Invoke-Expression "& '$FileName' --% $Arguments" 2>&1 -ErrorVariable errors | ForEach-Object {
         if ($_ -is [System.Management.Automation.ErrorRecord]) {
             Write-Error $_
-        } else {
+        }
+        else {
             Write-Host $_
         }
     }
 
-    foreach($errorMsg in $errors){
+    foreach ($errorMsg in $errors) {
         Write-Error $errorMsg
     }
     $ErrorActionPreference = 'Stop'
-    if($LASTEXITCODE -ne 0)
-    {
-         throw  (Get-VstsLocString -Key "SAD_AzureSQLDacpacTaskFailed" -ArgumentList $LASTEXITCODE)
+    if ($LASTEXITCODE -ne 0) {
+        throw  (Get-VstsLocString -Key "SAD_AzureSQLDacpacTaskFailed" -ArgumentList $LASTEXITCODE)
     }
 }
 
-function Detect-AuthenticationType
-{
-   param(
-     [String]$serverName,
-     [String]$databaseName,
-     [String]$sqlUsername,
-     [String]$sqlPassword,
-     [String]$aadSqlUsername,
-     [String]$aadSqlPassword,
-     [String]$connectionString
-   )
-
-   if ($serverName -and $databaseName) {
-     if ($sqlUsername -and $sqlPassword) {
-       $authenticationType = "server";
-     } elseif ($aadSqlUserName -and $aadSqlPassword) {
-       $authenticationType = "aadAuthenticationPassword";
-     } else {
-       $authenticationType = "aadAuthenticationIntegrated";
-     }
-   } elseif ($connectionString) {
-     $authenticationType = "connectionString";
-   } else {
-     throw (Get-VstsLocString -Key "SAD_InvalidAuthenticationInputs")
-   }
-
-   return $authenticationType
+function Detect-AuthenticationType {
```

Comment:
Please add the condition to return servicePrincipal authentication type.

Reply:
Good point but which condition makes the diference between aadAuthenticationIntegrated and servicePrincipal ?